radio/src/bluetooth.cpp

```diff
@@ -251,15 +251,15 @@ void Bluetooth::sendTrainer()
   int16_t PPM_range = g_model.extendedLimits ? 640*2 : 512*2;
 
   int firstCh = g_model.trainerData.channelsStart;
-  int lastCh = firstCh + 8;
+  int lastCh = firstCh + BLUETOOTH_TRAINER_CHANNELS;
 
   uint8_t * cur = buffer;
   bufferIndex = 0;
   crc = 0x00;
 
   buffer[bufferIndex++] = START_STOP; // start byte
   pushByte(TRAINER_FRAME);
-  for (int channel=0; channel<lastCh; channel+=2, cur+=3) {
+  for (int channel=firstCh; channel<lastCh; channel+=2, cur+=3) {
```

Comment:
Right here if you set the start channel higher than 0 it would screw up the data stream

Reply:
As in, this is now a problem (since before it was locked to 0)?

Reply:
Yeah looks like it always started at channel 0, even if you had firstCh set higher. If caused the data stream to be 

> > I did notice that the PPM polarity is backwards (EdgeTX Tx16s) & (OpenTX/Taranis) (Should it be flipped?)
> 
> do you mean “external module / PPM”? Or “Slave / Jack” Trainer output?



> > I did notice that the PPM polarity is backwards (EdgeTX Tx16s) & (OpenTX/Taranis) (Should it be flipped?)
> 
> do you mean “external module / PPM”? Or “Slave / Jack” Trainer output?

Slave/Jack output.

Reply:
If you wanted the start channel at channel 2 (to 10) then it would run the loop 10 times.. instead of the 8chs bluetooth is hard set for. Sending more data than expected. I just tested on Opentx too and it also does it wrong. In openTx you can set ch 1-4 and it still sends 8ch (also wrong). 

I'll take a look at fixing the B/W code too.. will double check @rotorman opinion first.

I feel like maybe this should be changed at some point to be able to send an arbitrary number of channels over bluetooth but so far everything I've seen is 8.

Reply:
I have no hw. to test Bluetooth slave operation, but if the channel range was not working before as displayed, this IMHO surely makes sense to fix. Thx!

About the trainer signal - I had a look at Futaba trainer jack with a scope and it has idle high, thus the way EdgeTX signal diagram in the first post on this page is doing. Thus, I believe this to be correct - but please double check in case I might be wrong.

Reply:
> If you wanted the start channel at channel 2 (to 10) then it would run the loop 10 times.. instead of the 8chs bluetooth is hard set for. 

Derp... of course it does... 🤦 Yup, that sounds good to me. 

With regard the the trainer signal, I don't think the question is whether it is right or wrong, but rather, if when setting to + polarity it being the exact opposite to OTX is intentional ;) And am I right in presuming it does correctly change if set to -?

Reply:
It does.